### PR TITLE
[backlog-10392] - parameter localizations engine/core

### PR DESCRIPTION
**Pull request by prakapenka**
Introduce new formula for resource lookup. We may consider to review ResourceLookupFunction.properties  for better display values.

### CODE REVIEW

**Comment by prakapenka** [comment]
Et dices ad Wingman

**Comment by prakapenka** [comment]
hey wingman check it again

File: engine/core/test-src/org/pentaho/reporting/engine/classic/core/modules/gui/base/parameters/TextFieldParameterComponentTest.java [hunk 1]
```diff
@@ -38,7 +38,6 @@
 import org.junit.Before;
 import org.junit.Test;
 import org.pentaho.reporting.engine.classic.core.ResourceBundleFactory;
-import org.pentaho.reporting.engine.classic.core.modules.gui.base.ParameterReportControllerPane;
 import org.pentaho.reporting.engine.classic.core.parameters.ParameterAttributeNames;
```
Comment (prakapenka) [inline engine/core/test-src/org/pentaho/reporting/engine/classic/core/modules/gui/base/parameters/TextFieldParameterComponentTest.java]: - import org.pentaho.reporting.engine.classic.core.modules.gui.base.ParameterReportControllerPane; is deleted while wingman still claim for

"/home/buildguy/project/engine/core/test-src/org/pentaho/reporting/engine/classic/core/modules/gui/base/parameters/TextFieldParameterComponentTest.javaorg.pentaho.reporting.engine.classic.core.modules.gui.base.ParameterReportControllerPane. at Line: 41 Column: 8"

wingman shows no mercy as usually.

**Comment by prakapenka** [comment]
wingman, wingman, preaching wing...

**Comment by prakapenka** [comment]
@tmorgner please review

**Comment by tmorgner** [comment]
Wingman says there is work to do.

**Comment by tmorgner** [comment]
How does this related to the other pull request #829 ? If that other one is obsolete now, remove it. 
If not, then you have overlapping changes and should bring both together or you get merge problems later on.

I also noticed that this PR here has no support for translating display-values of list-parameters. If that is intentional and you plan to add that later, then that is fine by me - just tell whether there will be another pull request or whether this PR will be extended.

So far, the code looks good. Nothing to complain - Wingman does a good job with that anyway :)

**Comment by prakapenka** [comment]
No need to translate display names via DefaultListValues since it has ComputedListValues and having for display values post-processing formula - we will get the same result just with no changes at all. Since we ok to translate error messages via foumlas expression - seems we ok to translate display values with post proc foumula. If we would have formulas for data format and for display label - we would write only resourcelookup formula and nothing else. Now we are going to define resource for label and data format in separate input. There also will be second PR to update designer UI immediately after this PR will be merged.

**Comment by prakapenka** [comment]
Wingman, the lord master of checkstyle purgatory, show me some mercy.

**Comment by tmorgner** [comment]
Waiting for Wingman, then merging ..

**Pull request merged**